django/db/models/sql/query.py

```diff
@@ -1430,11 +1431,15 @@ def build_lookup(self, lookups, lhs, rhs):
                 return
 
         lookup = lookup_class(lhs, rhs)
-        # Interpret '__exact=None' as the sql 'is NULL'; otherwise, reject all
-        # uses of None as a query value unless the lookup supports it.
+        # Interpret '__exact=None' as the SQL 'IS NULL'. For '__iexact=None' on
+        # KeyTransform, interpret it as '__exact=None' instead of 'IS NULL'
+        # (#36508). For all other cases, reject the use of None as a query
+        # value unless the lookup explicitly supports it.
         if lookup.rhs is None and not lookup.can_use_none_as_rhs:
             if lookup_name not in ("exact", "iexact"):
                 raise ValueError("Cannot use None as a query value")
+            if lookup_name == "iexact" and isinstance(lhs, KeyTransform):
+                return lhs.get_lookup("exact")(lhs, None)
```

Comment:
Having had another think, it would be preferable if we don't add json specific logic here
Can we not update `django.db.models.fields.json.KeyTransformIExact` somehow?

Reply:
I share your sentiment @sarahboyce, this should be implemented at the `KeyTransformIExact` level and not hardcoded here. The reason why `exact` and `iexact` are here today is due to how they must span `OUTER JOIN` when targeting a reverse relationship, I hope to remove it entirely with #16817.